Dockerfile

```diff
@@ -0,0 +1,13 @@
+FROM golang:latest AS golang
+WORKDIR /go/src/github.com/bitly/oauth2_proxy
+RUN go get -u github.com/golang/dep/cmd/dep
+COPY . .
+#RUN dep ensure && \
+RUN go get ./...
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o oauth2_proxy .
```

Comment:
SHOULD: Fix this.

The overloading of this Dockerfile is a very unusual pattern.  Usually the app image is not combined with the build image.  If we separated the two, the app image could get really small.

The app image could be really tiny, using alpine.  Using a larger image with dev tools installed increases the attack surface.

I would separate this out with a Makefile target that does a go build first, and then another Makefile target that does the docker build which copies the binary into an alpine image.  You need a makefile for a docker push anyways.

Reply:
Sweet.  Actually, it is a good thing.  Thanks for the link @josdotso 

> His Dockerfile adopts the multi-stage feature added recently to Docker: https://docs.docker.com/develop/develop-images/multistage-build/#before-multi-stage-builds`

Reply:
This is good.